backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/storage/lsm/LiveMigrateDiskCommand.java

```diff
@@ -304,6 +307,8 @@ public boolean performNextOperation(int completedChildCount) {
             return true;
         }
 
+        lockManager.releaseLock(removeSnapshotLock);
```

Comment:
If I'm not mistaken, since you `acquireLock()` only in `LIVE_MIGRATE_DISK_EXEC_COMPLETED` phase and release it always, there will be many `Trying to release exclusive lock which does not exist...` warnings in the `engine` logs?

Reply:
It gets to this line only in AUTO_GENERATED_SNAPSHOT_REMOVE_END phase

Reply:
Ahh, sorry, my bad.
But since it confused me, maybe better explicitly add this treatment in `AUTO_GENERATED_SNAPSHOT_REMOVE_END` phase.

Reply:
We can even change this method to `switch` before Benny comes back :)

Reply:
Agree, it would be more correct.

Reply:
not as part of this change please